specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzeconversations.json

```diff
@@ -1415,15 +1415,15 @@
         }
       }
     },
-    "Modality": {
+    "InputModality": {
```

Comment:
I thought calling the model this was a problem? I was only suggesting you set the `enum.name` to `InputModality` so the SDK uses that. I mean, if it matches the model even better, but I thought this was a problem.

Reply:
Let me ask @bidisha-c to take a look.

Reply:
adding @mikaelsitruk so comment on this.

Reply:
commented previously too, i would avoid changing the parameter name, the type can be changed but not the parameter. 
What is the rationale of changing the enum type? There is no really input/output modality so i don't understand why we need even to change the type.

Reply:
I misunderstood what exactly we need to update. As Heath commented out, we are going to update the enum name only so SDK can use it.

Reply:
The word "modality" (and one other I forget off hand) were too generic, which our SDK automation caught. *Maybe* we could get an exemption, though. @tg-msft @johanste?

Reply:
Looked it up. The other was "Modifier", though I don't remember what @ZulaMostafa decided to rename it in one of our SDK previews.

Reply:
@mikaelsitruk , any comments on Heath's comment on "modality" is too general?

Reply:
To clarify, too generic for use in the SDK. Since our code mixes with a lot of other frameworks and common libraries from language to language, we have [guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-problems-generic-words) and validation that avoids common, generic words. "Modality" and "modifiers" were caught by this validation. I don't see a problem keeping them in the REST API, but that's why I suggested either changing `enum.name` or adding `x-ms-class-name` if not an enum e.g., to a property. Then all autorest-based SDKs will use the same name automatically.

Reply:
I see. So the current shape looks good right? Modality is an enum, but the x-ms-enum name is InputModality.

Reply:
Correct. Just leave the swagger model as Modality and just change `x-ms-enum.name`.

Reply:
@tjprescott @akning-ms can the lint error be ignored going forward? Also, recommending "isO4217" for "ISO4217" seems like a bug. "IS" here isn't "is", but part of an acronym.

@lmazuel can you help override, or should we want to make sure we can override this?